web/frontend/src/schema/configurationValidation.ts

```diff
@@ -0,0 +1,400 @@
+      test() {
+        const { path, parent } = this;
+        const { MaxN, Choices, ID } = parent;
+
+        if (Choices.length !== MaxN) {
+          return this.createError({
+            path,
+            message: `Choices array length should be equal to MaxN in texts, in object ID: ${ID}`,
+          });
+        }
+        if (Choices.includes('')) {
+          return this.createError({
+            path,
+            message: `Choices array should not contain empty strings in texts, in object ID: ${ID}`,
+          });
+        }
+        return true;
+      },
+    })
+    .required(),
+});
+
+const subjectSchema = yup.object({
+  ID: yup.lazy(() => idSchema),
+  Title: yup.lazy(() => titleSchema),
+  Order: yup
+    .array()
+    .of(yup.string())
+    .test({
+      name: 'Testing Order',
+      message: 'Error Order array is not coherent with the Subject object',
+      test() {
+        const { path, parent } = this;
+        const { Order, Subjects, Ranks, Selects, Texts } = parent;
+        const onlyUnique = (value, index, self) => self.indexOf(value) === index;
+
+        // If we don't have unique IDs the array is not consistent with the Subject
+        if (Order.filter(onlyUnique).length !== Order.length) return false;
+
+        const subjectsID = Subjects.filter(onlyUnique).map((subject) => subject.ID);
+        const ranksID = Ranks.filter(onlyUnique).map((rank) => rank.ID);
+        const selectsID = Selects.filter(onlyUnique).map((select) => select.ID);
+        const textsID = Texts.filter(onlyUnique).map((text) => text.ID);
+        const allTypesID = [...subjectsID, ...ranksID, ...selectsID, ...textsID].filter(onlyUnique);
+
+        // Verify that the length of the Order array is exactly the length of the sum of the
+        // Subjects, Ranks, Selects and Texts arrays even when duplicates are removed
+        if (
+          Subjects.length + Ranks.length + Selects.length + Texts.length === Order.length &&
+          allTypesID.length === Order.length
+        ) {
+          let filteredOrder = [...Order];
+          for (const id of Order) {
+            // If we find the ID in any of the arrays we remove it from our filteredOrder array
+            if (
+              Subjects.find((subject) => subject.ID === id) ||
+              Ranks.find((rank) => rank.ID === id) ||
+              Selects.find((select) => select.ID === id) ||
+              Texts.find((text) => text.ID === id)
+            )
+              filteredOrder = filteredOrder.filter((order) => order !== id);
+          }
+          // If we found all the IDs of our Order in the arrays the test passes
+          // meaning that there is exactly one ID for each question type inside the Order array
+          if (filteredOrder.length === 0) return true;
+        }
+        return this.createError({
+          path,
+          message: `Error Order array is not coherent with the Subject in object ID: ${parent.ID}`,
+        });
```

Comment:
Here more than anywhere else we should return an error as soon as possible in order to provide the best context possible. In the actual implementation there is no way to know what specific ID is problematical for example. Also it seems that you don't need to have a filtered array:
```ts
        // Verify that the length of the Order array is exactly the length of the sum of the
        // Subjects, Ranks, Selects and Texts arrays even when duplicates are removed
        if (Subjects.length + Ranks.length + Selects.length + Texts.length === Order.length) {
          // return error
        }

        if (allTypesID.length === Order.length) {
          // return error
        }

        for (const id of Order) {
          // check if the ID corresponds to something, otherwise return an error
          if (
            !Subjects.find((subject) => subject.ID === id) &&
            !Ranks.find((rank) => rank.ID === id) &&
            !Selects.find((select) => select.ID === id) &&
            !Texts.find((text) => text.ID === id)
          ) {
            // return an error
          }
        }

        // return OK
      },
```